### ISSUE TITLE
Enable Inner Join across multiple embedding indexes 

### ISSUE THREAD

**Issue by csnelsonchu** [neuml/txtai#288]
Let's say I want to enable semantic search for a SQL database table named product.
The `product` table has columns `id`, `title`, `description` and `color`.
After I create 3 embedding index (`title`, `description` and `color`) with the same unique `id`
Would this be a possible to inner join the query in on inner join statement?
```
SELECT title.text, description.text, color.text, title.score, description.score, color.score
FROM title
INNER JOIN description ON (title.id = description.id)
INNER JOIN color ON (title.id = color.id)
WHERE title.text similar('shirt') OR description.text similar('winkle free') OR color.text similar('blue')
```

**Comment by davidmezzetti** [comment]
Thank you for the idea. Joins are currently not supported. txtai documents can have nested structures and supports a JSON like structure. So you could have a document like this.

```
description: text
color: text
title: text
```

and then combine standard filters with a similar clause. But as of now only a single text field is supported per index. It's possible this could be extended to support multiple ANN fields per index.

I'll keep this issue open but it's something that likely wouldn't be tackled until a 5.x or future release.

**Comment by csnelsonchu** [comment]
Thank you for confirming the roadmap of the project. It will be handy if it can be extended to multiple ANN fields per index. For now, I'll create a wrapper API to split traditional SQL query into multiple txtAI queries, so I can enable data analysts to create semantic search visualization reports.

**Comment by davidmezzetti** [comment]
I know this is an old issue but it's resolved with #515
